### PR TITLE
new experiment runner

### PR DESCRIPTION
**Pull request by BoltzmannBrain**
Run with e.g. `python experiments/experiment.py data/sample_reviews/sample_reviews_data_training.csv -m ClassificationModelFingerprint -mm fluent.models.classify_fingerprint --trainSize 1 10 20`

### CODE REVIEW

**Comment by BoltzmannBrain** [comment]
@scottpurdy please review. I'm particularly interested in your thoughts as to the architecture of the experiment; i.e. splitting it into a run script (experiment.py) and an experiment `Runner` class.

File: fluent/models/classification_model.py [hunk 1]
```diff
@@ -90,7 +85,55 @@ def logEncodings(self, patterns, path):
...
+    maxCount = 0
+    for c in labelCount:  ## TODO: better way to do this?
+      if c[1] > maxCount:
+        maxCount = c[1]
+    winners = [c[0] for c in labelCount if c[1]==maxCount]
+
+    return winners if len(winners) <= n else winners[:n]
+
+
+################################################################################
```
Comment (scottpurdy) [inline fluent/models/classification_model.py]: I prefer avoiding things like this. If it is really useful to separate these functions out then you can put them into a new module.

File: fluent/models/classification_model.py [hunk 2]
```diff
@@ -90,7 +85,55 @@ def logEncodings(self, patterns, path):
...
+# Evaluation methods
+
+  def calculateClassificationResults(self, classifications):  ## TODO: plot
+    """Calculate the classification accuracy for each category.
+    """
+    ## TODO
+
+
+  # def evaluateKFold(self, classifications, references, idx):
```
Comment (scottpurdy) [inline fluent/models/classification_model.py]: ?

File: fluent/models/classification_model.py [hunk 3]
```diff
@@ -166,10 +189,52 @@ def evaluateFinalResults(self, intermResults):
 
 
   @staticmethod
-  def _printTrialReport(labels, refs, idx):
+  def calculateAccuracy(classifications):   ## TODO: just get numpy arrays passed in
+  ## TODO: multiclass; this forces evaluation metrics to consider only the first predicted classification
```
Comment (scottpurdy) [inline fluent/models/classification_model.py]: If you're going to leave this at lease move it into the doc string

File: fluent/utils/plotting.py [hunk 4]
```diff
@@ -132,11 +132,11 @@ def plotCummulativeAccuracies():
   @staticmethod
   def interpretConfusionMatrixData(dataFrame, normalize):
     """Parse pandas dataframe into confusion matrix format."""
-    labels = dataFrame.columns.values.tolist()[1:-1]
+    labels = dataFrame.columns.values.tolist()[:-1]
     values = map(list, dataFrame.values)
-
+    import pdb; pdb.set_trace()
```
Comment (scottpurdy) [inline fluent/utils/plotting.py]: remove?

**Comment by scottpurdy** [comment]
:+1:

**Pull request merged**